sympy/controlsys/lti.py

```diff
@@ -0,0 +1,1363 @@
+        tfm to construct the state space model from, or the Matrices A,B,C,D as StateSpaceModel(A, B, C, D)
+
+    Examples
+    ========
+
+    >>> from sympy import Matrix, Symbol
+    >>> from sympy.controlsys.lti import StateSpaceModel, TransferFunctionModel
+
+    The easiest way to create a StateSpaceModel is via four matrices:
+
+    >>> A, B, C, D = Matrix([1,2]), Matrix([2,3]), Matrix([2]), Matrix([0])
+    >>> StateSpaceModel(A, B, C, D)
+    StateSpaceModel(
+    Matrix([
+    [1],
+    [2]]),
+    Matrix([
+    [2],
+    [3]]),
+    Matrix([[2]]),
+    Matrix([[0]]))
+
+    One can use less matrices as well. The rest will be filled with a minimum of
+    zeros:
+
+    >>> StateSpaceModel(A, B)
+    StateSpaceModel(
+    Matrix([
+    [1],
+    [2]]),
+    Matrix([
+    [2],
+    [3]]),
+    Matrix([[0]]),
+    Matrix([[0]]))
+
+    One can also use a TansferFunctionModel to create a StateSpaceModel
+
+    >>> s = Symbol('s')
+    >>> tfm = TransferFunctionModel(Matrix([1/s, s/(1 + s**2)]))
+    >>> StateSpaceModel(tfm)
+    StateSpaceModel(
+    Matrix([
+    [0, -1, 0],
+    [1,  0, 0],
+    [0,  1, 0]]),
+    Matrix([
+    [1],
+    [0],
+    [0]]),
+    Matrix([
+    [1, 0, 1],
+    [1, 0, 0]]),
+    Matrix([
+    [0],
+    [0]]))
+
+    See Also
+    ========
+
+    TransferFunctionModel: transfer function model of a lti system
+
+    References
+    ==========
+
+    Joao P. Hespanha, Linear Systems Theory. 2009.
+    """
+
+    def __init__(self, *arg):
+
+        def zero():
+            self.represent[0] = zeros(1)
+            self.represent[1] = zeros(1)
+            self.represent[2] = zeros(1)
+            self.represent[3] = zeros(1)
+
+        def one():
+            self.represent[1] = zeros(self.represent[0].shape[0], 1)
+            self.represent[2] = zeros(1, self.represent[0].shape[1])
+            self.represent[3] = zeros(1)
+
+        def two():
+            if not self.represent[0].shape[0] == self.represent[1].shape[0]:
+                raise ShapeError("Shapes of A,B,C,D must fit")
+            self.represent[2] = zeros(1, self.represent[0].shape[1])
+            self.represent[3] = zeros(1, self.represent[1].shape[1])
+
+        def three():
+            if not ((self.represent[0].shape[0] == self.represent[1].shape[0]) and
+                    (self.represent[0].shape[1] == self.represent[2].shape[1])):
+                raise ShapeError("Shapes of A,B,C,D must fit")
+            self.represent[3] = zeros(self.represent[2].shape[0],
+                                      self.represent[1].shape[1])
+
+        def default():
+            # assert that A,B,C,D have matching shapes
+            if not ((self.represent[0].shape[0] == self.represent[1].shape[0]) and
+                    (self.represent[0].shape[1] == self.represent[2].shape[1]) and
+                    (self.represent[1].shape[1] == self.represent[3].shape[1]) and
+                    (self.represent[2].shape[0] == self.represent[3].shape[0])):
```

Comment:
Clearly there is a better way to write this code

Reply:
Here, there is a pattern but coding it would probably create problems for other developers to understand the logic as it may lead non-intuitive code. So, I think for such small piece of code we can leave it as it is.